clang/lib/Sema/SemaChecking.cpp

```diff
@@ -19185,7 +19185,7 @@ static bool isLayoutCompatible(ASTContext &C, EnumDecl *ED1, EnumDecl *ED2) {
 
 /// Check if two fields are layout-compatible.
```

Comment:
Can you update this comment for the new member?  Is it both fields that are union members, or just one?

Reply:
Good catch! CWG2586 I implement here basically establish a different set of rules for layout compatibility of union members, and structs/classes are not layout compatible with unions by definition.

At this point I wonder if boolean is still needed, but my impression of our layout compatibility machinery is that it might be a bit too smart to infer the boolean. It's simple and recursive, without "looking back" so to say.

Reply:
After an offline discussion, we agreed to keep the boolean, and to add an additional assert to ensure that the boolean is used correctly.